django/core/management/commands/makemessages.py

```diff
@@ -126,13 +126,22 @@ def postprocess_messages(self, msgs):
         # Remove '.py' suffix
         if os.name == 'nt':
             # Preserve '.\' prefix on Windows to respect gettext behavior
-            old = '#: ' + self.work_path
-            new = '#: ' + self.path
+            old_path = self.work_path
+            new_path = self.path
         else:
-            old = '#: ' + self.work_path[2:]
-            new = '#: ' + self.path[2:]
+            old_path = self.work_path[2:]
+            new_path = self.path[2:]
 
-        return msgs.replace(old, new)
+        # matches a comment line with one or more files
+        pattern = '^(#:.+)(%s)(.*)$' % old_path
+        matches = re.findall(pattern, msgs, flags=re.MULTILINE)
+
+        for match in matches:
+            old = '%s%s%s' % match
+            new = '%s%s%s' % (match[0], new_path, match[2],)
```

Comment:
I'd chop the trailing comma.
Is it worth a comment explaining what the loop is doing? (it's not extremely obvious to me as someone not familiar with this code)